modules/space/rendering/renderableconstellation.cpp

```diff
@@ -0,0 +1,416 @@
+    , _constellationNamesFilename(ConstellationInfo)
+    , _constellationSelection(SelectionInfo)
+{
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+
+    addProperty(_opacity);
+    registerUpdateRenderBinFromOpacity();
+
+    _renderOption.addOption(RenderOptionViewDirection, "Camera View Direction");
+    _renderOption.addOption(RenderOptionPositionNormal, "Camera Position Normal");
+    // @TODO (abock. 2021-01-31) In the other classes, this is done with an enum, and
+    // doing it based on the fisheye rendering seems a bit brittle?
+    if (global::windowDelegate->isFisheyeRendering()) {
+        _renderOption = RenderOptionPositionNormal;
+    }
+    else {
+        _renderOption = RenderOptionViewDirection;
+    }
+    addProperty(_renderOption);
+
+    // Avoid reading files here, instead do it in multithreaded initialize()
+    if (p.constellationNamesFile.has_value()) {
+        _constellationNamesFilename =
+            absPath(p.constellationNamesFile.value().string()).string();
+    }
+    _constellationNamesFilename.onChange([&]() { loadConstellationFile(); });
+    addProperty(_constellationNamesFilename);
+
+    _lineWidth = p.lineWidth.value_or(_lineWidth);
+    addProperty(_lineWidth);
+
+    if (p.labelFile.has_value()) {
+        _labelFile = absPath(*p.labelFile).string();
+        _hasLabel = true;
+
+        _drawLabels = p.drawLabels.value_or(_drawLabels);
+        addProperty(_drawLabels);
+
+        _textColor = p.textColor.value_or(_textColor);
+        _hasLabel = p.textColor.has_value();
+        _textColor.setViewOption(properties::Property::ViewOptions::Color);
+        addProperty(_textColor);
+
+        _textOpacity = p.textOpacity.value_or(_textOpacity);
+        addProperty(_textOpacity);
+
+        _textSize = p.textSize.value_or(_textSize);
+        addProperty(_textSize);
+
+        _textMinMaxSize = p.textMinMaxSize.value_or(_textMinMaxSize);
+        _textMinMaxSize.setViewOption(properties::Property::ViewOptions::MinMaxRange);
+        addProperty(_textMinMaxSize);
+
+        if (p.labelUnit.has_value()) {
+            _labelUnit = codegen::map<DistanceUnit>(*p.labelUnit);
+        }
+        else {
+            _labelUnit = DistanceUnit::Meter;
+        }
+    }
+
+    _constellationSelection.onChange([this]() { selectionPropertyHasChanged(); });
+    addProperty(_constellationSelection);
+
+    _assetSelectedConstellations =
+        p.constellationSelection.value_or(_assetSelectedConstellations);
+}
+
+std::string RenderableConstellation::constellationFullName(
+                                                      const std::string& identifier) const
+{
+    if (_constellationNamesTranslation.empty() || identifier.empty()) {
+        std::string message = "List of constellations or the given identifier was empty";
+        LWARNINGC("RenderableConstellation", message);
+        return "";
+    }
+
+    try {
```

Comment:
We could replace this explicitly checking whether the `identifier` is in the `_constellationNamesTranslation` and skip the try/catch block